src/rdkafka_request.c

```diff
@@ -2283,9 +2284,11 @@ void rd_kafka_ConsumerGroupHeartbeatRequest(
         int16_t ApiVersion = 0;
         int features;
         size_t rkbuf_size = 0;
+        rd_kafkap_str_t *subscribed_topic_regex_to_send =
+            subscribed_topic_regex;
 
         ApiVersion = rd_kafka_broker_ApiVersion_supported(
-            rkb, RD_KAFKAP_ConsumerGroupHeartbeat, 0, 0, &features);
+            rkb, RD_KAFKAP_ConsumerGroupHeartbeat, 0, 1, &features);
```

Comment:
We can remove support for version 0 as we didn't have a GA version with it and avoid handling the case where we have a regex to send but the broker isn't supporting it.
```suggestion
            rkb, RD_KAFKAP_ConsumerGroupHeartbeat, 1, 1, &features);
```
In case it's not supported we can mark the coordinator as dead and return
```C
        if (ApiVersion == -1) {
                rd_kafka_cgrp_coord_dead(rkb->rkb_rk->rk_cgrp,
                                         RD_KAFKA_RESP_ERR__UNSUPPORTED_FEATURE,
                                         "ConsumerGroupHeartbeatRequest not "
                                         "supported by broker");
                return;
        }
```